src/pint/models/pulsar_binary.py

```diff
@@ -185,7 +187,31 @@ def apply_units(self):
             bparObj.value = bparObj.value * u.Unit(bparObj.units)
 
     def update_binary_object(self, toas=None, acc_delay=None):
-        """Update binary object instance for this set of parameters/toas."""
+        """Update stand alone binary's parameters and toas from PINT-facing object.
+
+        This function passes the PINT-facing object's parameter values and TOAs
+        to the stand-alone binary object. If the TOAs are not provided, it only
+        updates the parameters not the TOAs.
+
+        Parameters
+        ----------
+        toas: `~pint.toa.TOAs` object, optional
+            The TOAs that need to pass to the stand alone model.Default value is
+            None. If toas is None, this function only updates the parameter value.
+            If 'acc_delay' is not provided, the stand alone binary receives the
+            standard barycentered TOAs.
+
+        acc_delay: `~numpy.ndarray`
+            If provided, TOAs will be corrected by provided acc_delay instead of
+            the standard barycentering. The stand alone binary receives the
+            input TOAs - acc_delay.
```

Comment:
Just to be clear, this means that, if provided, the binary object will use the TDB MJDs minus whatever numbers are in `acc_delay`? How would one obtain sensible numbers for `acc_delay`? Can you give an example of how one would correctly use this parameter?

Reply:
acc_delay follows the APIs in the TimingModel.delay() method which uses the acc_delay. This is designed for the case if some delay depends on the total delay of previous delay components. One can use this parameter to pass it into the function like:
```
delay = all_other_delay(toas, acc_delay=None)
dependent_delay = dependent_delay_fun(toas, acc_delay=delay)
```
This is a very low-level function, most users should not have to use it. I agree with you this is not the most optimized code which I developed years ago. If we want to optimize this part. Let us do it in another PR.

Reply:
`acc_delay` appears in various places in our API, including in TimingModel itself; currently the documentation there is just `???`. Until we're ready to remove it it needs documentation, since (among others) your new test is relying on its behaviour.